### ISSUE TITLE
Update all matching tasks at once, and perhaps add a setting

### ISSUE THREAD

**Issue by reportaman** [tools-life/taskwiki#349]
As of now, if a tasks shows up on multiple lines in the document, the status of the last instance of it overrides the prior ones. This is all well and good, except in one case: let's say you had a hundred+ lines of tasks for a project. Imagine Task-A, and Task-B are both dependent on Task-C. So Task-C will appear in two places in the hundred+ lines of tasks. Let's assume Task-A is closer to the top of the list, and Task-B is somewhere below in the list. Now imagine you were going through subtasks of Task-A and updating their statuses. All of a sudden you realize that even if you update the status of Task-C, it reinstates its old value when you write the file ( ``:w<CR>``). Then you would have to search the last location of Task-C in the file, edit it and save the file again.

Proposal: When a ``<leader>t<char>`` key sequence is emitted, not only change the status of the task on the current line, but all the matching tasks in the file. This will make the UI much more cleaner. If this is something that does break some other way of using taskwiki, or if it is a performance concern, then add a setting to enable or disable this feature (for instance``g:taskwiki_updatematching``).

**Comment by reportaman** [comment]
Update: My example of multiple dependency is not valid, as taskwiki does not support it. But the same problem can arise if a user has multiple viewports, and a task shows in multiple of them. Even if you consider doing something like this, I guess its a low priority thing... I think I am settling down with a certain style of designing & editing my todo file, where this such problem will not arise.
